test/blackbox-tests/test-cases/ctypes/directories.t

```diff
@@ -0,0 +1,77 @@
+This test characterizes how the ctypes support deals with directories, in
+particular when relative paths are passed as `-I`.
+
+With `(using ctypes 0.2)`, some commands are executed in workspace root and
+others are executed in the directory where `(ctypes)` is found, so it is
+necessary to pass relative dirs twice. (this prevents vendorability, since this
+requires knowing the path from the workspace root to the current directory).
+
+To test that, we create a binding to `O_RDWR` and `open`, both in the C library.
+These are provided by `fcntl.h`, but instead of using this header, we're
+proxying it through a `lib.h` which should be in the include path for
+compilation to work.
+
+  $ cat > dune-project << EOF
+  > (lang dune 3.4)
+  > (using ctypes 0.2)
+  > EOF
+
+  $ mkdir lib
+
+  $ mk_dune () {
+  > local flags=$1
+  >  cat << EOF
+  > (executable
+  >  (name e)
+  >  (flags :standard -w -27)
+  >  (ctypes
+  >   (build_flags_resolver
+  >    (vendored (c_flags :standard $flags)))
+  >   (external_library_name l)
+  >   (headers (include lib.h))
+  >   (deps lib.h)
+  >   (type_description
+  >    (functor f)
+  >    (instance types))
+  >   (function_description
+  >    (functor f)
+  >    (instance functions))
+  >   (generated_entry_point entry)))
+  > EOF
+  > }
+
+  $ mk_dune "-I lib -I ." > lib/dune
+
+  $ cat > lib/lib.h << EOF
+  > #include <fcntl.h>
```

Comment:
Can we use a header that would be testable on windows? At least in theory. Something that's part of the C standard would be ideal.

Reply:
Ah, I was looking for something in the libc but forgot that it's not that standard. I'll try with something from stdlib.h or string.h.

Reply:
done using `BUFSIZ` and `fopen`.